sys/net/gnrc/network_layer/ipv6/nib/nib_ft.c

```diff
@@ -133,6 +133,7 @@ bool gnrc_ipv6_nib_ft_iter(const ipv6_addr_t *next_hop, unsigned iface,
         while ((offl = _nib_offl_iter(offl))) {
             assert(offl->mode != 0);
             if ((offl->next_hop != NULL) &&
+                (offl->mode != _PL || offl->flags & _PFX_ON_LINK) &&
                 ((iface == 0) || (iface == _nib_onl_get_if(offl->next_hop))) &&
                 ((next_hop == NULL) || ipv6_addr_equal(&offl->next_hop->ipv6,
                                                        next_hop))) {
```

Comment:
This if condition hurts my brain - how about some un-minification? 

```suggestion
            if ((offl->next_hop == NULL) {
                /* ??? */
                continue;
            }
            if (offl->mode == _PL && !(offl->flags & _PFX_ON_LINK)) {
                /* prefix list entry is off-link */
                continue;
            }
            if (iface && iface != _nib_onl_get_if(offl->next_hop) {
                /* interface does not match */
                continue;
            }
            if (next_hop && !ipv6_addr_equal(&offl->next_hop->ipv6, next_hop) {
                /* next hop does not match */
                continue;
            }
```

Reply:
Yes, this certainly improves readability!